en/docs/chapter_data_structure/summary.md

```diff
@@ -2,32 +2,65 @@
+From a storage perspective, the underlying structure of a hash table is an array, where each bucket might contain a value, a linked list, or a tree. Therefore, hash tables may contain both linear data structures (arrays, linked lists) and non-linear data structures (trees).
 
 **Q**: Is the length of the `char` type 1 byte?
 
-The length of the `char` type is determined by the encoding method used by the programming language. For example, Java, JavaScript, TypeScript, and C# all use UTF-16 encoding (to save Unicode code points), so the length of the char type is 2 bytes.
+The length of the `char` type is determined by the encoding method of the programming language. For example, Java, JavaScript, TypeScript, and C# all use UTF-16 encoding (to save Unicode code points), so the length of the `char` type is 2 bytes.
 
-**Q**: Is there ambiguity in calling data structures based on arrays "static data structures"? Because operations like push and pop on stacks are "dynamic".
+**Q**: Is there any ambiguity when we refer to array-based data structures as "static data structures"? The stack can also perform "dynamic" operations such as popping and pushing.
 
-While stacks indeed allow for dynamic data operations, the data structure itself remains "static" (with unchangeable length). Even though data structures based on arrays can dynamically add or remove elements, their capacity is fixed. If the data volume exceeds the pre-allocated size, a new, larger array needs to be created, and the contents of the old array copied into it.
+The stack can implement dynamic data operations, but the data structure is still "static" (the length is fixed). Although array-based data structures can dynamically add or remove elements, their capacity is fixed. If the stack size exceeds the pre-allocated size, then the old array will be copied into a newly created and larger array.
 
-**Q**: When building stacks (queues) without specifying their size, why are they considered "static data structures"?
+**Q**: When building a stack (queue), its size is not specified, so why are they "static data structures"?
 
-In high-level programming languages, we don't need to manually specify the initial capacity of stacks (queues); this task is automatically handled internally by the class. For example, the initial capacity of Java's ArrayList is usually 10. Furthermore, the expansion operation is also implemented automatically. See the subsequent "List" chapter for details.
+In high-level programming languages, we do not need to manually specify the initial capacity of stacks (queues); this task is automatically completed within the class. For example, the initial capacity of Java's `ArrayList` is usually 10. Furthermore, the expansion operation is also completed automatically. See the subsequent "List" chapter for details.
```

Comment:
optional (this will change the consistencies, but give more detail, so its up to you and Krahets):
the automatic capacity management can clarified further:
"For example, the initial capacity of Java's `ArrayList` is usually 10. Furthermore, the expansion operation is also completed automatically. See the subsequent "List" chapter for details."
->
"the initial capacity of Java's ArrayList is typically 10. However, it's important to note that while the class manages capacity automatically, the underlying implementation still uses a fixed-size array. When the data exceeds the current capacity, a new, larger array is created, and the old array's contents are copied over. This inherent fixed-size nature of the array-based implementation is why stacks (queues) are considered "static data structures." See the subsequent "List" chapter for more details."

Reply:
Thanks your pointing these out. I accept the first comment.

The second one is also good. But [this piece of text in the file](https://github.com/krahets/hello-algo/pull/1374/files#diff-c505c96ba9d28d8496bf8dc51f56793fc1acc59e0708656500a1fb322c590f11L29-R29) is very close to what you add.
Is it addressing your concern? If so, we do not need to add more discussion.